src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -168,7 +168,6 @@ def remove(self, name, handler):
     @staticmethod
     def _register_builtin_arguments(**kwargs):
         global_group = kwargs['global_group']
-        global_group.add_argument('--subscription', dest='_subscription_id', help=argparse.SUPPRESS)
```

Comment:
@tjprescott, I will wire it up differently when address #833. For now as we both know this flag is doing nothing